format-avro/src/test/java/io/cdap/plugin/format/avro/input/AvroInputFormatProviderTest.java

```diff
@@ -70,4 +71,68 @@ public void testSchemaDetection() throws IOException {
     Schema schema = schemaDetector.detectSchema(avroFile.getPath(), formatContext, Collections.emptyMap());
     Assert.assertEquals(expectedSchema, schema);
   }
+
+  @Test
+  public void testPathFieldAddition() throws IOException {
+    // write avro file
+    Schema expectedSchema = Schema.recordOf("Avro",
+                                            Schema.Field.of("id", Schema.of(Schema.Type.INT)),
+                                            Schema.Field.of("name", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+                                            Schema.Field.of("score", Schema.of(Schema.Type.DOUBLE)),
+                                            Schema.Field.of("pathField", Schema.of(Schema.Type.STRING)));
+    org.apache.avro.Schema avroSchema = new org.apache.avro.Schema.Parser().parse(expectedSchema.toString());
+    File avroFile = new File(TMP_FOLDER.newFolder(), "testPathFieldAddition.avro");
+    GenericRecord record = new GenericRecordBuilder(avroSchema)
+      .set("id", 0)
+      .set("name", "alice")
+      .set("score", 1.0d)
+      .set("pathField", avroFile.getAbsolutePath())
```

Comment:
This test shouldn't be writing anything for the path field in the data.

I'm confused how both this test and testPathFieldConflict() can be passing. They both have path field in the data.

Reply:
ok I see it is passing because detectSchema() returns the data schema when there is a path issue. The test should also be checking that there are no validation failures in the failure collector.

Reply:
Done